pkg/build/strategies/sti/sti.go

```diff
@@ -448,6 +448,44 @@ func (b *STI) Execute(command string, user string, config *api.Config) error {
 		NetworkMode:     string(config.DockerNetworkMode),
 	}
 
+	// If there are injections specified, override the original assemble script
+	// and wait till all injections are uploaded into the container that runs the
+	// assemble script.
+	if len(config.Injections) > 0 && command == api.Assemble {
+		injectedFiles, err := util.ExpandInjectedFiles(config.Injections)
+		if err != nil {
+			return err
+		}
+		rmScript, err := util.CreateInjectedFilesRemovalScript(injectedFiles)
+		if err != nil {
+			return err
+		}
+		defer os.Remove(rmScript)
+		glog.V(5).Infof("Waiting for injected files to be copied into assemble container...")
+		opts.CommandOverrides = func(cmd string) string {
+			return fmt.Sprintf("while [ ! -f %q ]; do sleep 0.5; done; %s; result=$?; source %s; exit $result",
+				"/tmp/rm-injections", cmd, "/tmp/rm-injections")
```

Comment:
no, if assemble returns non-zero then we still run injection cleanup... if
that returns non-zero we exit immediately. i updated the PR (set +e in the
script)

On Fri, Jan 8, 2016 at 5:54 PM, Rodolfo Carvalho <notifications@github.com>
wrote:

> In pkg/build/strategies/sti/sti.go
> <https://github.com/openshift/source-to-image/pull/380#discussion_r49210030>
> :
>
> > +	// and wait till all injections are uploaded into the container that runs the
> > +	// assemble script.
> > +	if len(config.Injections) > 0 && command == api.Assemble {
> > +		injectedFiles, err := util.ExpandInjectedFiles(config.Injections)
> > +		if err != nil {
> > +			return err
> > +		}
> > +		rmScript, err := util.CreateInjectedFilesRemovalScript(injectedFiles)
> > +		if err != nil {
> > +			return err
> > +		}
> > +		defer os.Remove(rmScript)
> > +		glog.V(5).Infof("Waiting for injected files to be copied into assemble container...")
> > +		opts.CommandOverrides = func(cmd string) string {
> > +			return fmt.Sprintf("while [ ! -f %q ]; do sleep 0.5; done; %s; result=$?; source %s; exit $result",
> > +				"/tmp/rm-injections", cmd, "/tmp/rm-injections")
>
> does that mean that if cmd returns non-zero exit code it will also exit
> immediately without sourcing/running /tmp/rm-injections?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/source-to-image/pull/380/files#r49210030>.
>



-- 
-----------------------------------------------
Michal Fojtik <mi@mifo.sk <mfojtik@mifo.sk>>
Red Hat OpenShift, Engineering